packages/utils/src/ts-eslint/Rule.ts

```diff
@@ -424,6 +425,19 @@ interface RuleListener {
   WithStatement?: RuleFunction<TSESTree.WithStatement>;
   YieldExpression?: RuleFunction<TSESTree.YieldExpression>;
 }
+type RuleListenerExitSelectors = {
+  [K in keyof RuleListenerBaseSelectors as `${K}:exit`]: RuleListenerBaseSelectors[K];
+};
+interface RuleListenerCatchAllBaseCase {
+  [nodeSelector: string]: RuleFunction | undefined;
+}
+// Interface to merge into for anyone that wants to add more selectors
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface RuleListenerExtension {}
+
+type RuleListener = RuleListenerBaseSelectors &
+  RuleListenerExitSelectors &
+  RuleListenerCatchAllBaseCase;
```

Comment:
https://github.com/typescript-eslint/typescript-eslint/pull/6679/files#diff-53e6896dd0f15121e29e2dfc7309a38d6d6637fc0a5a3bfe64bc4007ab5fca64R457

maybe we could add missing codePath functions to type?

Reply:
I just haven't bothered with them just because honestly I've never found an actual use for them in our lint rules.
Our users are apparently in the same boat because nobody has asked for them (317 dependents of our utils package listed in npm).

Maybe one day but I don't think it's worth messing up the DevX of node selectors for until we have an actual motivating usecase for.

Reply:
https://github.com/SonarSource/eslint-plugin-sonarjs/blob/4dfd992eeccafca92caed52889b4f83bb1e38c3c/src/rules/no-one-iteration-loop.ts#L41

one of examples :P